pkg/network/protocols/http2/protocol.go

```diff
@@ -343,6 +353,47 @@ func (p *protocol) GetStats() *protocols.ProtocolStats {
 	}
 }
 
+// UpdateKernelTelemetry should be moved to the HTTP/2 part as well
+func (p *protocol) UpdateKernelTelemetry(mgr *manager.Manager) {
+	var zero uint32
+
+	for {
+		mp, _, err := mgr.GetMap(probes.HTTP2TelemetryMap)
+		if err != nil {
+			log.Warnf("error retrieving http2 telemetry map: %s", err)
+			return
+		}
+
+		http2Telemetry := &HTTP2Telemetry{}
+		if err := mp.Lookup(unsafe.Pointer(&zero), unsafe.Pointer(http2Telemetry)); err != nil {
+			// This can happen if we haven't initialized the telemetry object yet
+			// so let's just use a trace log
+			if log.ShouldLog(seelog.TraceLvl) {
+				log.Tracef("error retrieving the http2 telemetry struct: %s", err)
+			}
+			return
+		}
+
+		p.http2Telemetry.http2requests.Add(int64(http2Telemetry.Request_seen))
+		p.http2Telemetry.http2responses.Add(int64(http2Telemetry.Response_seen))
+		p.http2Telemetry.endOfStreamEOS.Add(int64(http2Telemetry.End_of_stream_eos))
+		p.http2Telemetry.endOfStreamRST.Add(int64(http2Telemetry.End_of_stream_rst))
+		p.http2Telemetry.strLenExceedsFrame.Add(int64(http2Telemetry.Str_len_exceeds_frame))
+		p.http2Telemetry.pathSizeBucket0.Add(int64(http2Telemetry.Path_size_bucket0))
+		p.http2Telemetry.pathSizeBucket1.Add(int64(http2Telemetry.Path_size_bucket1))
+		p.http2Telemetry.pathSizeBucket2.Add(int64(http2Telemetry.Path_size_bucket2))
+		p.http2Telemetry.pathSizeBucket3.Add(int64(http2Telemetry.Path_size_bucket3))
+		p.http2Telemetry.pathSizeBucket4.Add(int64(http2Telemetry.Path_size_bucket4))
+		p.http2Telemetry.pathSizeBucket5.Add(int64(http2Telemetry.Path_size_bucket5))
+		p.http2Telemetry.pathSizeBucket6.Add(int64(http2Telemetry.Path_size_bucket6))
+		p.http2Telemetry.maxInterestingFrames.Add(int64(http2Telemetry.Max_interesting_frames))
+		p.http2Telemetry.maxFramesToFilter.Add(int64(http2Telemetry.Max_frames_to_filter))
```

Comment:
this logic should be encapsulated inside a method of `kernelTelemetry` 
for instance `func (t *kernelTelemetry) Update(tel *HTTP2Telemetry)` and `UpdateKernelTelemetry` should only call this method